Sources/JWTKeychain/Controllers/API/UserController.swift

```diff
@@ -66,13 +72,109 @@ open class UserController: UserControllerType {
 
     open func regenerate(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try JSON(node: ["token": token])
     }
 
     open func me(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try user.makeJSON(token: token)
     }
+
+    open func resetPasswordEmail(request: Request) -> ResponseRepresentable {
+        do {
+
+            guard
+                let email = request.data["email"]?.string,
+                let user: User = try User.query().filter("email", email).first() else {
+                    return JSON(["success": "Instructions were sent to the provided email"])
+            }
+
+            let token = try self.configuration.generateResetPasswordToken(user: user)
+
+            // Send mail
+            try Mailer.sendResetPasswordMail(drop: self.drop, user: user, token: token)
+            
+            return JSON(["success": "Instructions were sent to the provided email"])
+        } catch {
+            return JSON(["error": "An error occured."])
+        }
+    }
+
+    open func resetPasswordForm(request: Request, token: String) throws -> View {
+
+        // Validate token
+        if try !self.configuration.validateToken(token: token) {
+            throw Abort.notFound
+        }
+
+        let jwt = try JWT(token: token)
+
+        guard
+            let userId = jwt.payload["user"]?.object?["id"]?.int,
+            let _ = try User.query().filter("id", userId).first() else {
+            throw Abort.notFound
+        }
+
+        return try drop.view.make("ResetPassword/form", ["token": token])
```

Comment:
I might be misunderstanding, but this file is inside `/API/` - should we consider splitting this controller into two parts of maybe simply just removing the `API` folder - what do you think?